File: <100/063/063.jl
function is_ok(a :: Int, p :: Int) :: Bool
  a_ :: BigInt = a
  n = a_^p
  l = n |> string |> length
  return l == p
end


function main() :: Nothing
  answer = 0
  for a ∈ 1:9
    for p ∈ 1:100
      if is_ok(a, p)
        answer += 1
      else
        break
      end
    end
  end
  
  answer |> println
end


@time main()

File: <100/063/063.py
def is_ok(a: int, p: int) -> bool:
    n = a**p
    l = len(str(n))
    return l == p


def main() -> int:
    answer = 0
    for a in range(1, 10):
        for p in range(1, 100):
            if is_ok(a, p):
                answer += 1
            else:
                break
    return answer


if __name__ == "__main__":
    answer = main()
    print(answer)
